src/NBitcoin/ChainBase.cs

```diff
@@ -171,6 +228,11 @@ public IEnumerable<ChainedBlock> EnumerateAfter(uint256 blockHash)
             return this.EnumerateAfter(block);
         }
 
+        /// <summary>
+        /// Enumerates chain from the given block to tip.
+        /// </summary>
+        /// <param name="block">Block to enumerate from</param>
+        /// <returns>Enumeration of chain from given block to tip.</returns>
         public IEnumerable<ChainedBlock> EnumerateToTip(ChainedBlock block)
```

Comment:
"Enumeration of chain" sounds confusing to me (not only this method)
i'd be more happy with
"Enumeration of chained block headers from given block to the tip."

however, these methods seem extremely dangerous
i can see a lot of them being used in tests, in which case they should be migrated to tests where it is fine to have dangerous methods
but otherwise we should refactor if any other code is using those

again, not for this PR

Reply:
Yes I agree. I struggled with what to comment this as, I like your suggestion.